packages/hash/api/src/task-execution/index.ts

```diff
@@ -0,0 +1,36 @@
+import { DataSource } from "apollo-datasource";
+import fetch from "node-fetch";
+
+// TODO: When task scheduling is more mature and we move away from the temporary `hash-task-executor` we should have a single source of
+//  truth for available tasks, likely importable.
+export enum Tasks {
+  Demo = "python",
+}
+
+export type Config = {
+  host: string;
+  port: number;
+};
+
+export type TaskExecutor = ReturnType<typeof connectToTaskExecutor> &
+  DataSource;
+
+// A pretty dumb method at the moment, but provides a clean boundary where we can slot in logic for a more complex task-execution mechanism later on
+/**
+ *
+ * @param config The connection config for connecting to the Task-Executor
+ * @returns a connection object with a run_task method
+ */
```

Comment:
Nit:
```suggestion
/**
 * A pretty dumb method at the moment, but provides a clean boundary where we can slot in logic for a more complex task-execution mechanism later on
 * @param config The connection config for connecting to the Task-Executor
 * @returns a connection object with a run_task method
 */
```

Reply:
Would like to explore the thoughts on this a little. I was trying to convey two different things,

1. The **reasoning** for having such a silly and lightweight function (only useful when looking at the impl and wondering why it is the way it is)
2. The actual **documentation** of the `connectToTaskExecutor` interface, which **ideally** wouldn't change much when we add a more robust solution (although I suspect it will change)

Reply:
Ah I appreciate the distinction, I hadn't considered it when reviewing and read it as the documentation for the function. Happy for it to be left as is, I would've likely stuck this into a note at the end of the documentation, but this works as well. I don't feel strongly about this though to be clear and am happy for it to be left as is.
```suggestion

/**
 * @param config The connection config for connecting to the Task-Executor
 * @returns a connection object with a run_task method
 *
 * Note: a pretty dumb method at the moment, but provides a clean boundary where we can slot in logic for a more complex task-execution mechanism later on
 */
```